src/sys/sendfile.rs

```diff
@@ -36,30 +36,42 @@ cfg_if! {
     if #[cfg(any(target_os = "freebsd",
                  target_os = "ios",
                  target_os = "macos"))] {
-        use std::marker::PhantomData;
         use sys::uio::IoVec;
 
         #[allow(missing_debug_implementations)]
-        #[repr(C)]
-        pub struct SendfileHeaderTrailer<'a>(libc::sf_hdtr, PhantomData<&'a [IoVec<&'a [u8]>]>);
+        struct SendfileHeaderTrailer<'a>(
+            libc::sf_hdtr,
+            Option<Vec<IoVec<&'a [u8]>>>,
```

Comment:
Why is it necessary to store the `Option<Vec<` fields in `SendfileHeaderTrailer`?

Reply:
Because it's creating the `IoVec` structs from byte slices, it's now the logical owner of those structs.

Reply:
It still doesn't own the data, however.  `PhantomData` should be sufficient to keep the caller from dropping the byte slices.  Or is there something I'm missing?

Reply:
It doesn't own the byte slices, but it does own all of the `IoVec` instances, which do occupy memory.

Reply:
Ahh, that makes sense.  In that case, this PR LGTM.  All it needs is a squash.